src/main/java/com/_up/megastore/services/implementations/UserService.java

```diff
@@ -116,4 +117,77 @@ private User findUserToActivateOrThrowException(UUID userId,
             () -> new IllegalArgumentException("Token is invalid or user is already activated"));
   }
 
+  @Override
+  public void sendEmailToRecoverPassword(String email) {
+    if (!userRepository.existsByEmail(email))
+      throw new RuntimeException("User not found");
+
+    User user = findUserByEmailOrThrowException(email);
```

Comment:
Acá no hace falta que verifiquemos previamente si existe el usuario porque posteriormente lo obtenemos -y si no existe lanzamos una exception-

```suggestion
    User user = findUserByEmailOrThrowException(email);
```